packages/shorebird_cli/lib/src/logger.dart

```diff
@@ -1,8 +1,110 @@
+import 'dart:io';
+
 import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
 import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
 
 /// A reference to a [Logger] instance.
-final loggerRef = create(Logger.new);
+final loggerRef = create(ShorebirdLogger.new);
 
 /// The [Logger] instance available in the current zone.
-Logger get logger => read(loggerRef);
+ShorebirdLogger get logger => read(loggerRef);
+
+class ShorebirdLogger extends Logger {
+  /// {@macro shorebird_logger}
+  ShorebirdLogger({
+    Logger? logger,
+  }) : _logger = logger ?? Logger();
+
+  final Logger _logger;
```

Comment:
Since this extends `Logger` we don't need to inject an maintain an internal logger right?

Reply:
Right 🤡!